metriq-api/service/resultService.js

```diff
@@ -78,28 +142,20 @@ class ResultService extends ModelService {
     if (reqBody.method == null) {
       return { success: false, error: 'Result requires method to be defined.' }
     }
-    const method = await methodService.getByPk(reqBody.method)
-    if (!method) {
-      return { success: false, error: 'Result requires method to be present in database.' }
-    }
```

Comment:
Note that MethodService and PlatformService needed to be removed from ResultService in order to avoid circular dependencies, which prevent execution of the back end code. However, if the task/method/platform IDs don't exist, the back end will still refuse to create an invalid result row, just with a slightly less graceful error message.